src/MongoDB.Driver/Linq/Processors/OutOfCurrentScopePrefixCollector.cs

```diff
@@ -41,12 +42,25 @@ public static IEnumerable<string> Collect(Expression expression)
         public override Expression Visit(Expression node)
         {
             var hasOutOfCurrentScopePrefix = node as IHasOutOfCurrentScopePrefix;
-            if (hasOutOfCurrentScopePrefix != null && !string.IsNullOrWhiteSpace(hasOutOfCurrentScopePrefix.OutOfCurrentScopePrefix))
+            if (hasOutOfCurrentScopePrefix != null &&
+                !string.IsNullOrWhiteSpace(hasOutOfCurrentScopePrefix.OutOfCurrentScopePrefix) &&
+                !_isParentConstantExpression)
             {
                 _outOfCurrentScopePrefixCollection.Add(hasOutOfCurrentScopePrefix.OutOfCurrentScopePrefix);
             }
 
             return base.Visit(node);
         }
+
+        protected internal override Expression VisitWhere(WhereExpression node)
+        {
+            if (node.Source.NodeType == ExpressionType.Constant)
+            {
+                _isParentConstantExpression = true;
```

Comment:
Now, we have a flag about whether a source of a `where` expression is a `constantExpression` or not.

Reply:
Does the name of this field need to be more accurate?

Perhaps: `_isWhereSourceConstantExpression`?

`_isParentConstantExpression` is misleading.

Reply:
done